src/OptionSettlementEngine.sol

```diff
@@ -290,77 +290,71 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
     }
 
     /// @inheritdoc IOptionSettlementEngine
-    /// @dev Supplying claimId as 0 to the overloaded write signifies that a new
-    /// claim NFT should be minted for the options lot, rather than being added
-    /// as an existing claim.
-    function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        return write(optionId, amount, 0);
-    }
-
-    /// @inheritdoc IOptionSettlementEngine
-    function write(uint256 optionId, uint112 amount, uint256 claimId) public returns (uint256) {
-        (uint160 optionKey, uint96 decodedClaimNum) = decodeTokenId(optionId);
-
-        // optionId must be zero in lower 96b for provided option Id
-        if (decodedClaimNum != 0) {
-            revert InvalidOption(optionId);
-        }
-
-        // claim provided must match the option provided
-        if (claimId != 0 && ((claimId >> 96) != (optionId >> 96))) {
-            revert EncodedOptionIdInClaimIdDoesNotMatchProvidedOptionId(claimId, optionId);
-        }
-
+    function write(uint256 tokenId, uint112 amount) public returns (uint256) {
+        // You need to write some amount
         if (amount == 0) {
             revert AmountWrittenCannotBeZero();
         }
 
+        // Get the optionKey and claimNum from the tokenId
+        (uint160 optionKey, uint96 claimNum) = decodeTokenId(tokenId);
+
+        // Pass through the tokenId as the encodedClaimId
```

Comment:
We could add a further explanation in your comment — "In the case this is not an existing claim, this will be set on L324" or something 

* not blocking